editor/editor_inspector.cpp

```diff
@@ -2664,6 +2678,17 @@ Button *EditorInspector::create_inspector_action_button(const String &p_text) {
 	return button;
 }
 
+EditorInspector::ThemeCache &EditorInspector::get_singleton_theme_cache() {
+	if (!theme_singleton) {
+		theme_singleton = memnew(EditorInspector);
+		theme_singleton->hide();
+		if (EditorNode::get_singleton()) {
+			EditorNode::get_singleton()->get_gui_base()->add_child(theme_singleton);
+		}
```

Comment:
This condition fixed the crash, but technically causes a leak.

Reply:
Can we add it as child somewhere else as fallback to make sure it will get freed?

I have to say it feels a bit hacky to keep a static shadow Inspector for theme caching, isn't that pretty wasteful in number of objects and memory used to basically store a config file?

Reply:
It's still better than storing the cache in each inspector property individually. Unfortunately the main inspector is initialized too late, so it couldn't be used. Also, the fake inspector is empty, so it doesn't take that much.

> Can we add it as child somewhere else as fallback to make sure it will get freed?

SceneTree does not exist at all in this situation, so no.

Reply:
So what should we do? I'm not too fond of knowingly adding a leak. It's not for the main usage of the editor so it's not too bad but it would still be better without introducing a technical "bug".

Reply:
I used a deferred static method.